packages/clarity-js/src/performance/interaction.ts

```diff
@@ -0,0 +1,124 @@
+// This calculations is inspired by the web-vitals implementation, for more details: https://github.com/GoogleChrome/web-vitals
+import { PerformanceEventTiming, Interaction } from '@clarity-types/data';
+
+// Estimate variables to keep track of interactions
+let interactionCountEstimate = 0;
+let minKnownInteractionId = Infinity;
+let maxKnownInteractionId = 0;
+
+let prevInteractionCount = 0; // Used to track interaction count between pages
+
+const MAX_INTERACTIONS_TO_CONSIDER = 10; // Maximum number of interactions we consider for INP
+const DEFAULT_DURATION_THRESHOLD = 40; // Threshold to ignore very short interactions
+
+// List to store the longest interaction events
+const longestInteractionList: Interaction[] = [];
+// Map to track interactions by their ID, ensuring we handle duplicates
+const longestInteractionMap: Map<number, Interaction> = new Map();
+
+/**
+ * Update the approx number of interactions estimate count if the interactionCount is not supported.
+ * The difference between `maxKnownInteractionId` and `minKnownInteractionId` gives us a rough range of how many interactions have occurred. 
+ * Dividing by 7 helps approximate the interaction count more accurately, since interaction IDs may not always increase sequentially.
+ */
+const countInteractions = (entry: PerformanceEventTiming) => {
+  if ('interactionCount' in Performance) return;
+
```

Comment:
imo it'd be a bit clearer to have this function put Performance.interactionCount into interactionCountEstimate, then getInteractionCount can just always return interactionCountEstimate || 0 (and you don't have to check for interactionCount in performance twice or cast anything to number